packages/frontend-plugin-api/src/extensions/TranslationBlueprint.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createExtensionBlueprint } from '../wiring';
+import { createTranslationExtension } from './createTranslationExtension';
+import { TranslationMessages, TranslationResource } from '../translation';
+
+export const TranslationBlueprint = createExtensionBlueprint({
+  kind: 'translation',
```

Comment:
Hmm, yep feels like `namespace: params => ...` in the blueprint definition could be an alright way forward. I've been kinda tempted to unwrap some of these creators and for example be able to define the translation resource at the same time as creating the extension. That won't work with blueprints though, since we can't declare and infer any additional type parameters for `.make`. That leaves us with with deriving options from the params most likely being the best option.